### PR TITLE
Ensure x-request-id header matches kafka key

### PR DESCRIPTION
**Pull request by cliveseldon**
- [x] Ensure we set x-request-id kafka header when we create a new kafka produce in pipelinegateway thus ensuring a new one won't be generated
- [x] make kafka headers lowercase when converting from http headers which will be in canonical form, e.g. X-Request-ID -> x-request-id

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/v2/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by sakoush — APPROVED** [review]

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *[sakoush](https://github.com/SeldonIO/seldon-core/pull/4511#pullrequestreview-1222000878 "Approved")*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/v2/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: scheduler/pkg/kafka/pipeline/utils.go
Comment (agrski) [inline scheduler/pkg/kafka/pipeline/utils.go:67]: :upside_down_face: Minor point, but we only need to convert the form of the key if we're going to propagate it

Comment (cliveseldon) [reply scheduler/pkg/kafka/pipeline/utils.go:67]: We do propogate it and x-request-id (lowercase) is used amongst others,. When these are converted back to http later if needed they will revert to case when added.

**Pull request merged**
